var/spack/repos/builtin/packages/acts-algebra-plugins/package.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class ActsAlgebraPlugins(CMakePackage):
+    """A portable linear algebra library with multiple backends that is part of
+    the Acts ecosystem."""
+
+    homepage = "https://github.com/acts-project/algebra-plugins"
+    url = "https://github.com/acts-project/algebra-plugins/archive/refs/tags/v0.24.0.tar.gz"
+    list_url = "https://github.com/acts-project/algebra-plugins/tags"
+
+    maintainers("stephenswat")
+
+    license("MPL-2.0")
+
+    version("0.24.0", sha256="f44753e62b1ba29c28ab86b282ab67ac6028a0f9fe41e599b7fc6fc50b586b62")
+
+    variant(
+        "cxxstd",
+        default="17",
+        values=("17", "20", "23"),
+        multi=False,
+        description="C++ standard used",
+    )
+    variant("eigen", default=False, description="Enables the Eigen plugin")
+    variant("smatrix", default=False, description="Enables the SMatrix plugin")
+    variant("vecmem", default=False, description="Enables the vecmem plugin")
+    variant("vc", default=False, description="Enables the Vc plugin")
+
+    depends_on("cmake@3.14:", type="build")
+    depends_on("vecmem@0.22.0:", when="+vecmem")
+    depends_on("eigen@3.4.0:", when="+eigen")
+    depends_on("vc@1.4.0:", when="+vc")
+    depends_on("root@6.18.0:", when="+smatrix")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("CMAKE_CXX_STANDARD", "cxxstd"),
+            self.define("ALGEBRA_PLUGINS_USE_SYSTEM_LIBS", True),
+            self.define_from_variant("ALGEBRA_PLUGINS_INCLUDE_EIGEN", "eigen"),
```

Comment:
I was a bit confused by the CMakeLists.txt. This would seem to require also the `ALGEBRA_PLUGINS_SETUP_EIGEN3` (default False) since otherwise `find_package` is never called. This is also true for others, e.g. `Vc`.

Reply:
Yeah, unfortunately a lot of our CMake systems are unnecessarily complex; you are right that this will need to be set.

Reply:
I have added separate `INCLUDE` and `SETUP` variables for these.